src/network/stream_reader.rs

```diff
@@ -68,6 +68,10 @@ impl<R: Read> StreamReader<R> {
                         return Err(encode::Error::Io(io::Error::from(io::ErrorKind::UnexpectedEof)));
                     }
                 },
+                Err(encode::Error::UnrecognizedNetworkCommand(message, len)) => {
+                    self.unparsed.drain(..len);
```

Comment:
Just a note for future readers: This can't panic because for the `UnrecognizedNetworkCommand` error to be returned the payload length has to already have been verified in `CheckedData::consensus_decode`.